src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java

```diff
@@ -0,0 +1,96 @@
+package de.retest.recheck.ui.descriptors;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ui.image.Screenshot;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class RootElementFilter {
+	private final Filter filter;
+
+	public RootElement filter( final RootElement rootElement ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return rootElement;
+		}
+		final IdentifyingAttributes newIdentifyingAttributes =
+				filter( rootElement, rootElement.getIdentifyingAttributes() );
+		final Attributes newAttributes = filter( rootElement, rootElement.getAttributes() );
+		final String retestId = rootElement.getRetestId();
+		final Screenshot screenshot = rootElement.getScreenshot();
+		final String screen = rootElement.getScreen();
+		final int screenId = rootElement.getScreenId();
+		final String title = rootElement.getTitle();
+		final List<Element> containedElements = rootElement.getContainedElements();
+
+		final RootElement newRootElement = new RootElement( retestId, newIdentifyingAttributes, newAttributes,
+				screenshot, screen, screenId, title );
+
+		final List<Element> newContainedElements = filterContainedElements( newRootElement, containedElements );
+		newRootElement.addChildren( newContainedElements );
+		return newRootElement;
+	}
+
+	public IdentifyingAttributes filter( final Element element, final IdentifyingAttributes identifyingAttributes ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return identifyingAttributes;
+		}
+		final List<Attribute> attributes = identifyingAttributes.getAttributes();
+
+		final Collection<Attribute> filteredAttributes = attributes.stream() //
+				.filter( attribute -> !filter.matches( element, attribute.getKey() ) ) //
+				.collect( Collectors.toList() );
+		return new IdentifyingAttributes( filteredAttributes );
+	}
+
+	public Attributes filter( final Element element, final Attributes attributes ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return attributes;
+		}
+		final MutableAttributes filteredAttributes = new MutableAttributes();
+		final Iterator<Entry<String, Object>> iterator = attributes.iterator();
```

Comment:
Instead of an iterator, you can a for loop:

```java
for ( final Entry<String, Object> attribute : attributes ) {
	if ( filter.matches( element, attribute.getKey() ) ) {
		continue;
	}
	filteredAttributes.put( attribute.getKey(), (String) attribute.getValue() );
}
```

Or like elsewhere with a stream:

```java
StreamSupport.stream( attributes.spliterator(), false ) //
		.filter( attribute -> !filter.matches( element, attribute.getKey() ) ) //
		.forEach( attribute -> filteredAttributes.put( attribute.getKey(), (String) attribute.getValue() ) );
);
```